.github/templates/common.yml.j2

```diff
@@ -109,3 +109,55 @@ concurrency:
           fetch-depth: 0
           submodules: !{{ submodules }}
 {%- endmacro -%}
+
+
+{%- macro upload_test_reports(name) -%}
+      - name: Zip test reports for upload
+        if: always()
+        env:
+{%- if name == 'linux' or name == 'windows' %}
+          FILE_SUFFIX: '${{ github.job }}-${{ matrix.config }}-${{ matrix.shard }}-${{ matrix.num_shards }}-${{ matrix.runner }}'
```

Comment:
Since we're doing a lot of conditional logic here would it make sense to just make these two separate functions?

Reply:
it's a bit of a mess but not something we can cleanly separate without overdoing it since there are 4 types of branches taken. This is mostly useful in that we can concentrate all the report uploading logic in a single place